hepcrawl/parsers/crossref.py

```diff
@@ -0,0 +1,387 @@
+
+    @property
+    def url(self):
+        return self.record.get("URL")
+
+    @property
+    def material_source(self):
+        return self.record.get("source")
+
+    @property
+    def material(self):
+        title = self.title
+        subtitle = self.subtitle
+        if 'erratum' in title.lower() or 'erratum' in subtitle.lower():
+            material = 'erratum'
+        elif 'addendum' in title.lower() or 'addendum' in subtitle.lower():
+            material = 'addendum'
+        elif "publisher's note" in title.lower() or "publisher's note" in subtitle.lower():
+            material = 'addendum'
+        else:
+            material = 'publication'
+
+        return material
+
+    @property
+    def publication_info(self):
+        publication_info = {
+            'artid': self.artid,
+            'journal_title': self.journal_title,
+            'journal_issue': self.journal_issue,
+            'journal_volume': self.journal_volume,
+            'page_start': self.page_start,
+            'page_end': self.page_end,
+            'year': self.year,
+            'material': self.material,
+            'parent_isbn': self.parent_isbn,
+        }
+
+        return publication_info
+
+    @property
+    def parent_isbn(self):
+        isbn = self.record.get("ISBN")
+        if isbn:
+            return isbn[0]
+        else:
+            return isbn
+
+    @property
+    def journal_title(self):
+        j_title = self.record.get("short-container-title")
+
+        if j_title == []:
+            return None
+        else:
+            return j_title[0]
+
+    @property
+    def artid(self):
+        return self.record.get("article-number")
+
+    @property
+    def journal_issue(self):
+        return self.record.get("issue")
+
+    @property
+    def journal_volume(self):
+        return self.record.get("volume")
+
+    @property
+    def year(self):
+        date_parts = self.record.get("issued")
+
+        if date_parts:
+            date = date_parts.get("date-parts")[0][0]
+            if date:
+                return date
+            else:
+                date_parts = self.record.get("created")
+                if date_parts:
+                    date = date_parts.get("date-parts")[0][0]
+                    if date:
+                        return date
+        return None
+
+    @property
+    def page_start(self):
+        pages = self.record.get("page")
+
+        if pages:
+            return pages.split('-')[0]
+        else:
+            return None
+
+    @property
+    def page_end(self):
+        pages = self.record.get("page")
+
+        if pages:
+            return pages.split('-')[1]
+        else:
+            return None
+
+    @property
+    def number_of_pages(self):
+        first_page = self.page_start
+        last_page = self.page_end
+
+        if first_page and last_page:
+            return int(self.page_end) - int(self.page_start) or 1
+
+    @staticmethod
+    def get_author_name(author_node):
```

Comment:
it should be called `author_key` instead of `author_node` everywhere, as we are dealing with a `dict`, not XML.